stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/PopUtil.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.email.util;
+
+import org.ballerinalang.jvm.BallerinaValues;
+import org.ballerinalang.jvm.types.BArrayType;
+import org.ballerinalang.jvm.types.BTypes;
+import org.ballerinalang.jvm.values.ArrayValue;
+import org.ballerinalang.jvm.values.ArrayValueImpl;
+import org.ballerinalang.jvm.values.MapValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.mail.Address;
+import javax.mail.BodyPart;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMultipart;
+
+/**
+ * Contains Utility functions related to the POP3 protocol.
+ */
+public class PopUtil {
+
+    private static final Logger log = LoggerFactory.getLogger(PopUtil.class);
+    private static final int MAX_PORT = 65535;
+
+    /**
+     * Generates Properties object using the passed MapValue.
+     *
+     * @param popConfig MapValue with the configuration values
+     * @param isPop True if the protocol is POP3 and false otherwise (if protocol is IMAP)
+     * @return Properties object
+     */
+    public static Properties getProperties(MapValue popConfig, boolean isPop) {
```

Comment:
IMO, a method should have single responsibility. So, the implementation logic change based on a boolean means, you can have 2 separate methods.

Reply:
We have decided to move on with a single email retrieval client for the moment. We will later decide to split it into to two.

Reply:
Yes. We can divide logic into two Java methods for more clarity.

Reply:
Done.